src/app/projects/add-edit-project/add-edit-project.component.ts

```diff
@@ -791,81 +850,46 @@ export class AddEditProjectComponent implements OnInit, AfterViewInit, OnDestroy
   }
 
   /**
-   * Publish the selected logos.
+   * Publish the selected logos or shapefiles. Logos and shapefiles are already saved to the DB
+   * when selected in the file picker.
    *
    * @returns {void}
    */
-  private publishSelectedLogos() {
+  private publishAttachedFiles() {
     const logoValues = this.getLogosFormValues();
-    const documentPublishRequests = logoValues.map(logo => {
-      return this.documentService.publish(logo.document);
+    const shapefileValues = this.getShapefilesFormValues();
+    // Filters down to the IDs of the documents/files to publish
+    const logoDocuments = logoValues.map(logo => logo.document);
+    const shapefileDocuments = shapefileValues.map(shapefile => shapefile.document);
+    const allDocuments = logoDocuments.concat(shapefileDocuments);
+    
+    const documentPublishRequests = allDocuments.map(doc => {
+      return this.documentService.publish(doc);
     });
+
     forkJoin(documentPublishRequests)
     .subscribe(
       aggregateResponse => {
         aggregateResponse.forEach((individualResponse: Document|HttpErrorResponse) => {
           if ("status" in individualResponse) {
-            // One or more of the responses is an error.
+            // If one or more of the responses is an error.
             if (500 === individualResponse.status || 400 === individualResponse.status) {
               console.error('Error publishing file', individualResponse);
-              alert('There was a problem publishing one or more of the selected logos.')
+              alert('There was a problem publishing one or more of the selected logos/shapefiles.')
```

Comment:
Just an optional nitpick but you could just say 'files' to cover your bases in the future.

Reply:
In spite of the function name, it still only does apply to logos and shapefiles. My take would that by keeping the error more specific, we help the user or us troubleshoot things better